### PR TITLE
Nav bar item highlight feature added | issue #378

### LINKED ISSUE #378: Multiple UI Issues/Improvements
# Nav bar item does not highlight current selected item!

## Steps to reproduce
<li>Click on any nav bar item, eg leaderboard, contributors, projects etc. 
<li>You will see that it does not stays highlighted

## Expected behavior
<li>It should stay highlighted

image

# Nav bar does not close on clicking item in phone view

## Steps to reproduce 
<li>Click on any nav bar item in phone view
<li>You will see it does not close the side bar!

## Expected behavior
<li>Side bar/Nav bar should close on clicking on any item in phone view. 

image

### PR DESCRIPTION
**Pull request by Samuel-Aktar-Laskar**
# Closes #378 

## Shows selected item in nav bar
Now highlights selected item in nav bar in both dark, light mode. Small and Large screen. 
image
image
image
image

## In phone screen, on clicking item in sidebar, it closes the sidebar.

### CODE REVIEW

**Comment by rithviknishad** [comment]
Also, 

1. Clear the merge conflicts
2. It'd be great if you could link the issue in the PR comment using closing tags.
3. Give a meaningful title to the PR that conveys what this PR does. So that it's easily readable when we look at the changelogs in the main branch.

**Comment by rithviknishad** [comment]
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by Samuel-Aktar-Laskar** [comment]
> Hey nice work. Just few suggestions....


Thank you very much for the suggestions. I will resolve them all by tomorrow.

File: components/navbar/Navbar.tsx [hunk 1]
```diff
@@ -68,12 +73,19 @@ export default function Navbar() {
...
             <div className="flex flex-col items-center justify-center gap-2 md:px-4 md:py-2">
-              {Object.entries(MenuItems).map(([href, label]) => (
+              {Object.entries(MenuItems).map(([href, label], index) => (
                 <Link
-                  onClick={() => setOpen(!open)}
-                  key={href}
+                  key={index}
+                  onClick={() => {
+                    setOpen(!open);
+                  }}
```
Comment (rithviknishad) [inline components/navbar/Navbar.tsx:81]:
```suggestion
                  key={href}
                  onClick={() => setOpen(!open)}
```

File: components/navbar/Navbar.tsx [hunk 2]
```diff
@@ -68,12 +73,19 @@ export default function Navbar() {
               </button>
             </div>
             <div className="flex flex-col items-center justify-center gap-2 md:px-4 md:py-2">
-              {Object.entries(MenuItems).map(([href, label]) => (
+              {Object.entries(MenuItems).map(([href, label], index) => (
```
Comment (rithviknishad) [inline components/navbar/Navbar.tsx:76]:
```suggestion
              {Object.entries(MenuItems).map(([href, label]) => (
```

File: components/navbar/Navbar.tsx [hunk 3]
```diff
@@ -19,19 +19,24 @@ const MenuItems = {
...
+  const pathname = usePathname();
   return (
     <>
       <nav className="sticky top-0 z-10 border-b border-secondary-300 bg-background px-4 py-1 shadow-md dark:border-secondary-700 sm:shadow-lg">
         <div className="mx-auto flex max-w-7xl items-center justify-between xl:px-3">
           <Logo />
 
           <div className="hidden flex-row items-center justify-between gap-3 rounded bg-secondary-100 font-semibold dark:bg-secondary-800 md:rounded-full md:px-6 md:py-1 lg:flex">
-            {Object.entries(MenuItems).map(([href, label]) => (
+            {Object.entries(MenuItems).map(([href, label], index) => (
```
Comment (rithviknishad) [inline components/navbar/Navbar.tsx:30]:
```suggestion
            {Object.entries(MenuItems).map(([href, label]) => (
```

File: components/navbar/Navbar.tsx [hunk 4]
```diff
@@ -19,19 +19,24 @@ const MenuItems = {
...
               <Link
-                key={href}
-                className="text-sm transition-all hover:text-primary-500 hover:underline hover:dark:text-primary-300 md:p-2 md:text-base"
+                key={index}
```
Comment (rithviknishad) [inline components/navbar/Navbar.tsx:32]:
```suggestion
                key={href}
```

**Review by rithviknishad — APPROVED** [review]

**Comment by bodhish** [comment]
@Samuel-Aktar-Laskar, Nice work, lets get this merged!

**Pull request merged**